.github/workflows/mutexbot.yml

```diff
@@ -0,0 +1,122 @@
+name: MutexBot Container Image
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "mutexbot/**"
+  push:
+    branches:
+      - main
+    paths:
+      - "mutexbot/**"
+    tags:
+      - "mutexbot-v*.*.*"
+
+permissions:
+  contents: read
+
+env:
+  DOCKERHUB_REPO: neondatabase/dev-actions
+
+jobs:
+  build:
+    runs-on: ${{ fromJson(format('["self-hosted", "{0}"]', matrix.platform == 'linux/arm64' && 'small-arm64' || 'small')) }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - linux/amd64
+          - linux/arm64
+    steps:
+      - name: Fetch mutexbot folder
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: mutexbot
+
+      - name: Prepare
+        run: |
+          platform=${{ matrix.platform }}
+          echo "PLATFORM_PAIR=${platform//\//-}" >> $GITHUB_ENV
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.DOCKERHUB_REPO }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.NEON_DOCKERHUB_USERNAME }}
+          password: ${{ secrets.NEON_DOCKERHUB_PASSWORD }}
+
+      - name: Build and push by digest
+        id: build
+        uses: docker/build-push-action@v6
+        with:
+          context: mutexbot
+          platforms: ${{ matrix.platform }}
+          labels: ${{ steps.meta.outputs.labels }}
+          outputs: type=image,name=${{ env.DOCKERHUB_REPO }},push-by-digest=true,name-canonical=true,push=true
+
+      - name: Export digest
+        run: |
+          mkdir -p ${{ runner.temp }}/digests
+          digest="${{ steps.build.outputs.digest }}"
+          touch "${{ runner.temp }}/digests/${digest#sha256:}"
+
+      - name: Upload digest
+        uses: actions/upload-artifact@v4
+        with:
+          name: digests-${{ env.PLATFORM_PAIR }}
+          path: ${{ runner.temp }}/digests/*
+          if-no-files-found: error
+          retention-days: 1
+
+  merge:
+    runs-on: ["self-hosted", "small"]
+    needs:
+      - build
+    steps:
+      - name: Download digests
+        uses: actions/download-artifact@v4
+        with:
+          path: ${{ runner.temp }}/digests
+          pattern: digests-*
+          merge-multiple: true
```

Comment:
Can we set job output and use digests from it instead of using artifacts?

Reply:
switched to using outputs instead of artifacts